pkg/operator/deploy/deploy.go

```diff
@@ -131,9 +131,10 @@ func (o *operator) resources() ([]runtime.Object, error) {
 				Name: arov1alpha1.SingletonClusterName,
 			},
 			Spec: arov1alpha1.ClusterSpec{
-				ResourceID: o.oc.ID,
-				ACRName:    o.env.ACRName(),
-				Location:   o.env.Location(),
+				ResourceID:      o.oc.ID,
+				ACRName:         o.env.ACRName(),
+				Location:        o.env.Location(),
+				DevelopmentMode: o.env.IsDevelopment(),
```

Comment:
yaaaaaaaaay :P

Reply:
I would really really really like to avoid our development artefacts being exposed to customer visible code. Lets try removing this. It ok to have these checks in the code, but I think this is bit too far :/

Reply:
the main issue here is validating the VmSize that is in dev mode but not in prod. I'll maybe move to a container ENV variable

Reply:
+1.  I am wondering whether we should reuse RP_MODE for this (even though the naming is terrible).  Not sure.